### DISCUSSION TITLE
Help starting with Ebiten

### DISCUSSION THREAD

**Discussion by Mirudhula-m** [hajimehoshi/ebiten#1851] [Q&A] [answered]
Hello,

I am very new to both Golang and Ebiten, so please excuse me if this question is very basic. 
I tried to implement one of the example programs but I keep getting the following error on my terminal:

```
cannot find package "github.com/hajimehoshi/ebiten/v2" in any of:
	/usr/local/go/src/github.com/hajimehoshi/ebiten/v2 (from $GOROOT)
	/Users/RookieKoala/go/src/github.com/hajimehoshi/ebiten/v2 (from $GOPATH)
main.go:7:2: cannot find package "github.com/hajimehoshi/ebiten/v2/examples/resources/images" in any of:
	/usr/local/go/src/github.com/hajimehoshi/ebiten/v2/examples/resources/images (from $GOROOT)
	/Users/RookieKoala/go/src/github.com/hajimehoshi/ebiten/v2/examples/resources/images (from $GOPATH)
```

The lines of code which stops the program are:

```
import (
	"github.com/hajimehoshi/ebiten/v2"
	"github.com/hajimehoshi/ebiten/v2/examples/resources/images"
)
```

My GOROOT is set to /usr/local/go.
And GOPATH is set to $HOME/go.

I can see that the problem arises because there is no 'v2' folder inside the folder 'ebiten'. But, this is what the example code had. So, I removed the v2 from the line but I am getting a different error as follows:

```
../../../../../go/src/github.com/hajimehoshi/ebiten/internal/glfw/callback_notwindows.go:21:2: cannot find package "github.com/go-gl/glfw/v3.3/glfw" in any of:
	/usr/local/go/src/github.com/go-gl/glfw/v3.3/glfw (from $GOROOT)
	/Users/RookieKoala/go/src/github.com/go-gl/glfw/v3.3/glfw (from $GOPATH)
```

This might be trivial but please help me through what the problem might be.

Thank you!

**Comment by hajimehoshi** [comment, marked as answer]
Hi,

Did you follow the installation instruction: https://ebiten.org/documents/install.html ? Probably you are missing `go.mod` or something.

Thanks,

**Reply by Mirudhula-m** [reply]
My GO111MODULE is set to off, so up until now I have never used go mod. Is it required to be on in order to use this package?

**Reply by hajimehoshi** [reply]
Yes, I think everyone should use Go modules whenever possible.

**Reply by hajimehoshi** [reply]
In addition, Ebiten v2 doesn't work with GOPATH unfortunately.

**Reply by Mirudhula-m** [reply]
Ahh got it, it's working now! I was trying to see if I can use it without turning on the. GO111MODULE. I will avoid doing that then. Thanks for your help!
